analysis/monkwindwalker/src/modules/tier/PrimordialPotential.tsx

```diff
@@ -54,48 +57,58 @@ class PrimordialPotential extends Analyzer {
       this.onCast,
     );
     this.addEventListener(
-      Events.damage.by(SELECTED_PLAYER).spell(DAMAGE_AFFECTED_BY_PRIMORDIAL_POWER),
+      Events.damage
+        .by(SELECTED_PLAYER | SELECTED_PLAYER_PET)
+        .spell(DAMAGE_AFFECTED_BY_PRIMORDIAL_POWER),
       this.onDamage,
     );
   }
 
   onCast(event: CastEvent) {
-    if (!this.selectedCombatant.hasBuff(SPELLS.PRIMORDIAL_POWER_BUFF.id)) {
-      return;
+    if (this.selectedCombatant.hasBuff(SPELLS.PRIMORDIAL_POWER_BUFF.id)) {
+      this.wasLastCastPowered.add(event.ability.guid);
+
+      this.numberPoweredCasts += 1;
+
+      if (!this.poweredCasts[event.ability.guid]) {
+        this.poweredCasts[event.ability.guid] = 0;
+      }
+
+      this.poweredCasts[event.ability.guid] += 1;
+
+      event.meta = event.meta || {};
+      event.meta.isEnhancedCast = true;
+      const reason = (
+        <>
+          This cast was empowered by <SpellLink id={SPELLS.PRIMORDIAL_POWER_BUFF.id} /> (4-set
+          bonus)
+        </>
+      );
+      event.meta.enhancedCastReason = event.meta.enhancedCastReason ? (
+        <>
+          {event.meta.enhancedCastReason}
+          <br />
+          {reason}
+        </>
+      ) : (
+        reason
+      );
+    } else {
+      this.wasLastCastPowered.delete(event.ability.guid);
     }
```

Comment:
flip the `if` on this. generally prefer early return over wrapping an entire method in an `if`.

```suggestion
    if (!this.selectedCombatant.hasBuff(SPELLS.PRIMORDIAL_POWER_BUFF.id)) {
      this.wasLastCastPowered.delete(event.ability.guid);
      return;
    }

    this.wasLastCastPowered.add(event.ability.guid);

    this.numberPoweredCasts += 1;

    if (!this.poweredCasts[event.ability.guid]) {
      this.poweredCasts[event.ability.guid] = 0;
    }

    this.poweredCasts[event.ability.guid] += 1;

    event.meta = event.meta || {};
    event.meta.isEnhancedCast = true;
    const reason = (
      <>
        This cast was empowered by <SpellLink id={SPELLS.PRIMORDIAL_POWER_BUFF.id} /> (4-set
        bonus)
      </>
    );
    event.meta.enhancedCastReason = event.meta.enhancedCastReason ? (
      <>
        {event.meta.enhancedCastReason}
        <br />
        {reason}
      </>
    ) : (
      reason
    );
```